src/components/preview/preview.js

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+function Preview (props) {
+  const isLoading = (() => {
+    if (typeof props.loading !== 'undefined') {
+      return props.loading;
+    }
+    return !props.children;
+  })();
+
+  const mainClasses = (() => {
+    return classNames(
+      'carbon-preview',
+      { 'carbon-preview--placeholder': isLoading },
+      props.className
+    );
+  })();
+
+  return (
+    <div className={ mainClasses }>
+      { props.children }
+    </div>
+  );
+}
+
+Preview.propTypes = {
+  /**
+  * Children elements
+  *
+  * @property children
+  * @type {Node}
+  */
+  children: PropTypes.node,
+  /**
+  * Custom className
+  *
+  * @property className
+  * @type {String}
+  */
+  className: PropTypes.string,
```

Comment:
We don't normally allow setting custom CSS classes on components. Is this really needed or can we remove it?

Reply:
I have seen this on several component. Toast for example. Isn't this to allow override the css.